File: trab.cpp
#include <bits/stdc++.h>
using namespace std;



class Estacao{
    public:
        string nome;
        int x;
        int y;
        Estacao(string nome, int x, int y){
            this->nome = nome;
            this->x = x;
            this->y = y;
        }
};

class Linha{
    public:
        int n;
        string cor;
        vector<Estacao> estacoes;
};

map<string, pair<int, int>> estacoes;
vector<Estacao> adj[25]; //25 numero de estações

void lerEstacoes(){
    ifstream in_estacoes("in_estacoes.txt");
    string nome;
    while(getline(in_estacoes, nome, '>')){
        int x, y;
        in_estacoes >> x >> y;
        estacoes[nome] = {x, y};
        in_estacoes.ignore();
    }
}

void lerLinhas(int linha, string cor, int n){
    ifstream in_linhas;
    for(int i=0; i<n; i++){
        string estacao;
        int x;
        int y;
        getline(in_linhas, estacao, '>');
        //estacao = estacao.substr(1, estacao.size());
        cout << "estacao: " << estacao;
        //x = estacoes[estacao].second.first;
        //y = estacoes[estacao].second.second;
        //Estacao e = new Estacao(estacao, x , y);
        //adj[linha].push_back(new Estacao(estacao, x , y));
    }
    string teste;
    getline(cin, teste);
    cout << "teste: " << teste;
}
int main()
{
    //int linhas[7] = {1, 2, 4, 5, 6, 9, 11};
    lerEstacoes();
    for(int i=0; i<7; i++){
        string nomeLinha;
        string cor;
        int n;
        //cin >> nomeLinha >> cor >> n;
        //lerLinhas(i, cor, n);
    }
            for(auto x : estacoes){
            cout << "nome: " << x.first << " x: " << x.second.first << " y: " << x.second.second << "\n";
        }

}

File: trab.py
import matplotlib.pyplot as plt
import numpy as np


fig, ax = plt.subplots()
ax.set_xlim(0, 100)
ax.set_ylim(0, 100)

class Estacao:
    #nome;
    #x;
    #y;

    def __init__(self, nome, x, y):
        self.nome = nome
        self.x = x
        self.y = y

cores=['#ffbf00', '#0039e6', '#b300b3', '#ff6600', '#339933', '#739900', '#663300'] #linhas cores
ind = {} #ind das estacoes
estacoes = {} #mapa das estacoes

adj = []


def lerEstacoes():
    in_estacoes = open('in_estacoes.txt', 'r');
    count = 0
    coord_x = []
    coord_y = []
    nomes = []
    for line in in_estacoes:
        nome = line.split('>')[0]
        x = line.split('>')[1]
        y = line.split('>')[2]
        estacao = Estacao(nome, int(x), int(y))
        ind[nome] = count
        estacoes[nome] = estacao
        adj.append([])
        coord_x.append(int(x))
        coord_y.append(int(y))
        nomes.append(nome)
        count +=1
    
    ax.scatter(coord_x, coord_y, marker='o')
    for i, txt in enumerate(nomes):
        ax.text(coord_x[i]+1, coord_y[i], txt, fontsize=9)

def lerLinhas():
    in_linhas = open('in_linhas.txt', 'r');
    i = 0
    #7 pois sao 7 linhas
    while(i<7):
        line = in_linhas.readline().split()
        nome = line[0]
        cor = line[1]
        j = int(line[2])
        coord_x = []
        coord_y = []
        while(j>0):
            linha = in_linhas.readline().split('>')
            ind1 = ind.get(linha[0])
            #ind2 = ind.get(linha[1])
            estacao1 = estacoes.get(linha[0])
            estacao2 = estacoes.get(linha[1].replace('\n',''))
            adj[ind1].append(estacao2)
            coord_x.append(estacao1.x)
            coord_y.append(estacao1.y)
            if(j==1):
                coord_x.append(estacao2.x)
                coord_y.append(estacao2.y)
            j-=1
        ax.plot(coord_x, coord_y, color=cores[i])
        i +=1

lerEstacoes()
lerLinhas()
plt.show()